node/node.go

```diff
@@ -349,6 +350,23 @@ func (n *Node) newCommitValidator() p2p.GossipValidator {
 	}
 }
 
+// newFraudProofValidator returns a pubsub validator that validates a fraud proof and forwards
+// it to be verified
+func (n *Node) newFraudProofValidator() p2p.GossipValidator {
+	return func(fraudProofMsg *p2p.GossipMessage) bool {
+		n.Logger.Debug("fraud proof received", "from", fraudProofMsg.From, "bytes", len(fraudProofMsg.Data))
+		var fraudProof types.FraudProof
+		err := fraudProof.UnmarshalBinary(fraudProofMsg.Data)
+		if err != nil {
+			n.Logger.Error("failed to deserialize fraud proof", "error", err)
+			return false
+		}
+		// TODO(manav): Add validation checks for fraud proof here
```

Comment:
This should be tracked somewhere else than just in a todo.

Reply:
This [issue](https://github.com/celestiaorg/rollmint/issues/549) covers it.